### PR TITLE
[Cypress Updates] App analytics updates

### LINKED ISSUE #1886: [BUG] Failing Tests in the Observability CI
## Failing Tests in the Observability CI

We have a list of tests in the dashboards-observability repo that have been falky for a long time. Therefore, creating this issue, to track fixes made to this CI. Some the of tests haven't been updated with the change in code and something like FTR CI action may be deeper investigation as they run well on the FTR repo. Below are the CI actions that need a fix. 

 
Screenshot 2024-06-10 at 11 08 47 AM

### PR DESCRIPTION
**Pull request by ritvibhatt**
### Description
1. Update cypress tests for app analytics 
2. Update to use savedObjects client to delete saved visualizations in applications
3. Add back visualization action menu

Screenshot 2024-11-21 at 4 50 00 PM

Visualization action menu:
Screenshot 2024-12-04 at 3 01 06 PM

### Issues Resolved
#1886 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ps48 — APPROVED** [review]

File: .cypress/integration/app_analytics_test/app_analytics.spec.js [hunk 1]
```diff
@@ -432,11 +435,12 @@ describe('Viewing application', () => {
     cy.get('[data-test-subj="app-analytics-configTab"]').click();
     cy.get('select').select(visOneName);
     cy.intercept('PUT', `**/api/observability/application`).as('selectUpdate');
-    cy.wait('@selectUpdate')
-    cy.wait(2000); // despite the previous wait grabbing the call that updates the panel select, it doesn't appear without this
```
Comment (paulstn) [inline .cypress/integration/app_analytics_test/app_analytics.spec.js:436]: Do we know why the tests now work without this?

File: .cypress/integration/app_analytics_test/app_analytics.spec.js [hunk 2]
```diff
@@ -479,8 +483,7 @@ describe('Separate from other plugins', () => {
     cy.visit(
       `${Cypress.env('opensearchDashboards')}/app/observability-dashboards#/`
     );
-    cy.get('[data-test-subj="operationalPanelsActionsButton"]', { timeout: timeoutDelay }).click();
-    cy.get('[data-test-subj="addSampleContextMenuItem"]', { timeout: timeoutDelay }).click();
+    cy.get('.euiButtonContent').contains('Add samples').click();
```
Comment (paulstn) [inline .cypress/integration/app_analytics_test/app_analytics.spec.js:486]: Does this element have a `data-test-subj`?

**Review by paulstn — COMMENTED** [review]
Great work fixing the tests! Regarding the changes to the component files, are there also some bug fixes and features in this PR? If so, could you put them in the description?

File: .cypress/integration/app_analytics_test/app_analytics.spec.js (hunk 1 above)
Comment (ritvibhatt) [reply .cypress/integration/app_analytics_test/app_analytics.spec.js:436]: The test now reloads the home page after changing the availability which also makes the applications update properly

File: .cypress/integration/app_analytics_test/app_analytics.spec.js (hunk 2 above)
Comment (ritvibhatt) [reply .cypress/integration/app_analytics_test/app_analytics.spec.js:486]: It does not

**Review by TackAdam — APPROVED** [review]

**Pull request merged**